### PR TITLE
Cleanup diagram toolbox

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?
When a new diagram item was created, one had to add it to `gaphor.ui.diagramtoolbox`.

### What is the new behavior?

Toolbox functionality is located closer to the actual `Item`.

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: gaphor/diagram/diagramtools.py [hunk 1]
```diff
@@ -158,7 +159,8 @@ class PlacementTool(_PlacementTool):
     PlacementTool is used to place items on the canvas.
     """
 
-    def __init__(self, view, item_factory, after_handler=None, handle_index=-1):
+    # TODO: I would expect ElementFactory to be provided as well.
```
Comment (danyeaw) [inline gaphor/diagram/diagramtools.py]: Is there anything we need to do to close out this TODO?

Comment (amolenaar) [reply gaphor/diagram/diagramtools.py]: Currently the element factory is accessed though `diagram.model` (`view.canvas.diagram.model` to be precise). I think this is not ideal, and violates the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). I wanted to wait for this to fix until most of the code is in place.

I'm a bit occupied with Advent of Code at the moment, though :D.

Comment (amolenaar) [reply gaphor/diagram/diagramtools.py]: It's not as bad as I thought. I think using just `diagram.model` is fine. It makes the code less cluttered at least.

**Comment by danyeaw** [comment]
:tada:

**Pull request merged**